src/main/java/org/folio/event/service/AuditOutboxService.java

```diff
@@ -81,77 +82,97 @@ public Future<Integer> processOutboxEventLogs(Map<String, String> okapiHeaders)
     );
   }
 
+  private List<Future<Boolean>> getKafkaFutures(List<OutboxEventLog> eventLogs, Map<String, String> okapiHeaders) {
+    return eventLogs.stream().map(eventLog -> {
+      switch (eventLog.getEntityType()) {
+        case ORDER -> {
+          PurchaseOrder entity = Json.decodeValue(eventLog.getPayload(), PurchaseOrder.class);
+          OrderAuditEvent.Action action = OrderAuditEvent.Action.fromValue(eventLog.getAction());
+          return producer.sendOrderEvent(entity, action, okapiHeaders);
+        }
+        case ORDER_LINE -> {
+          PoLine entity = Json.decodeValue(eventLog.getPayload(), PoLine.class);
+          OrderLineAuditEvent.Action action = OrderLineAuditEvent.Action.fromValue(eventLog.getAction());
+          return producer.sendOrderLineEvent(entity, action, okapiHeaders);
+        }
+        case PIECE -> {
+          Piece entity = Json.decodeValue(eventLog.getPayload(), Piece.class);
+          PieceAuditEvent.Action action = PieceAuditEvent.Action.fromValue(eventLog.getAction());
+          return producer.sendPieceEvent(entity, action, okapiHeaders);
+        }
+        default -> throw new IllegalArgumentException();
```

Comment:
Found this now, need to change with more specific exception.

Reply:
or maybe to not complicate just add some error message to the constructor